react/components/GoTopButton/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import { Button } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import styles from './styles.css'
+
+interface Props {
+  topPixel?: number
+}
+
+const GoTopButton: StorefrontFC<Props> = ({ topPixel = 1000 }) => {
+  const [bodyOffset, setBodyOffset] = useState(
+    document.body.getBoundingClientRect()
```

Comment:
```suggestion
    () => document.body.getBoundingClientRect()
```

Anyway, this guarantees this function is only run once